### PR TITLE
[doc] cleanup references and links

### PR DESCRIPTION
**Pull request by picnixz**
We did not run `make linkcheck` on our own doc. There are some broken links but those are because of expired SSL certificates. Should I remove/ignore them or not?

```
examples.rst:73: [broken] https://anh.cs.luc.edu/python/hands-on/3.1/handsonHtml/: HTTPSConnectionPool(host='anh.cs.luc.edu', port=443): Max retries exceeded with url: /python/hands-on/3.1/handsonHtml/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: 
CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)')))

examples.rst:380: [broken] https://opendylan.org/: HTTPSConnectionPool(host='opendylan.org', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)')))

examples.rst:353: [timeout] https://www.adobe.com/devnet-docs/acrobatetk/: HTTPSConnectionPool(host='www.adobe.com', port=443): Read timed out. (read timeout=5)
```

The Adobe link is working but I think it's a matter of weird javascript and we would only be able to properly check with a headless browser. As for opendylan and anh.cs.luc.edu, should I remove them? I removed dead links and updated those that I could about (with the exception of Elemental which is no more maintained).

### CODE REVIEW

**Comment by jayaddison** [comment]
Maybe a too-complicated idea, for links like these that seem like they might be intermittent failures, but we're not sure...

We could manually add the port number with two leading zeroes if it is not in the URL, to indicate that it's flaky: `https://opendylan.org:00443/` for example.

The linkcheck should be run by a different person next time -- to reduce the chance of local config/network conditions biasing results.  If the link returns `working`, then remove a leading zero.  If it fails (`broken` or `timeout`), add another leading zero.

**Comment by jayaddison** [comment]
(if it reaches quorum - maybe five leading zeroes?  then we remove it entirely.  if no leading zeroes remain, then we remove the port number segment entirely, unless it is a non-standard-port)

**Comment by picnixz** [comment]
> We could manually add the port number with two leading zeroes if it is not in the URL, to indicate that it's flaky: [https://opendylan.org:00443/](https://opendylan.org/) for example.

Do you mean that your linkcheck does not fail? (the website is not flaky and I can access it normally from firefox; not sure if there is some caching causing the issue).

- For adobe, we cannot do anything I think.
- For anh, the certificate is expired so HTTP on port 80 fixes the issue.

**Comment by jayaddison** [comment]
> Do you mean that your linkcheck does not fail? (the website is not flaky and I can access it normally from firefox; not sure if there is some caching causing the issue).

I don't know yet, but any links in the main branch where the netloc contains `:0`, I'll re-check soon.

**Comment by picnixz** [comment]
I think I'll merge this one. We can check links separately (at least, with this PR, dead links are removed). By the way, the dead links were manually checked.

**Comment by picnixz** [comment]
By the way, on HEAD, the opendylan opens normally. So I don't know what happens with the linkcheck builder for this specific link.

File: doc/conf.py [hunk 1]
```diff
@@ -101,11 +101,22 @@
 latex_use_xindy = True
 
 linkcheck_timeout = 5
+linkcheck_ignore = [
+    r'^contents\.html$',  # extra generated page
+    '^\.\./contents\.html$',
+    re.escape('https://gitlab.com/projects/new'),  # requires sign-in
+    re.escape('https://web.libera.chat/?channel=#sphinx-doc'),
```
Comment (jayaddison) [inline doc/conf.py:108]: @picnixz Could you double-check that these are correctly filtering these links?  When I ran this locally I saw a `broken` result for the `#sphinx-doc` channel when I expected an `ignored` result.

(sorry, late reply)

Comment (picnixz) [reply doc/conf.py:108]: It's weird because I did not have the issue on my side. I'll try it again in a few moment

Comment (picnixz) [reply doc/conf.py:108]: image

For me it's ignored. And in the json output:

```json
{
  "filename": "support.rst",
  "lineno": 4,
  "status": "ignored",
  "code": 0,
  "uri": "https://web.libera.chat/?channel=#sphinx-doc",
  "info": ""
}
```

Comment (jayaddison) [reply doc/conf.py:108]: Ok, great - thanks; confirmed here too.

```
(           index: line    4) -ignored- https://web.libera.chat/?channel=#sphinx-doc 
```

Comment (jayaddison) [reply doc/conf.py:108]: (reason for the filename diff: I moved it temporarily to a near-empty `index` file)

**Pull request merged**